File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.cs
public class Solution
{
    public string solution(string[] survey, int[] choices)
    {
        string answer = "";
        string type = "RTCFJMAN";
        int[] count = new int[8] { 0, 0, 0, 0, 0, 0, 0, 0 };

        for (int i = 0; i < survey.Length; i++)
        {
            count[type.IndexOf(survey[i][0])] += (4 - choices[i]);
        }

        for (int i = 0; i < 8; i += 2)
        {
            answer += (count[i] >= count[i + 1]) ? type[i] : type[i + 1];
        }

        return answer;
    }
}

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    cnt = [0, 0, 0, 0]
    t = ['RT', 'CF', 'JM', 'AN']
    
    for sv, ch in zip(survey, choices):
        if sv in t:
            cnt[t.index(sv)] += ch-4
        else:
            cnt[t.index(sv[::-1])] += 4-ch
            
    return ''.join(t[i][cnt[i] > 0] for i in range(4))